tests/test_relation.py

```diff
@@ -98,6 +101,25 @@ def test_insert_select(self):
             'real_id', 'date_of_birth', 'subject_notes', subject_id='subject_id+1000', species='"human"'))
         assert_equal(len(self.subject), 2*original_length)
 
+    @raises(InternalError)
+    def test_insert_select_ignore_extra_fields0(self):
```

Comment:
Include a test case where the relation you are inserting from is not a simple table but rather a result of a query.